### PR TITLE
Improve desugar implementation for websocket auth

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
$subject

Related to: https://github.com/ballerina-platform/ballerina-lang/pull/27894
Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/1405

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java [hunk 1]
```diff
@@ -88,47 +95,51 @@
...
-    boolean isHttpPackage(List<BType> expressionTypes) {
+    void desugarFunction(BLangFunction functionNode, SymbolEnv env, List<BType> expressionTypes) {
+        if (isValidPackage(expressionTypes, HTTP_PACKAGE_NAME)) {
+            addHttpAuthDesugarFunctionInvocation(functionNode, env);
+        } else if (isValidPackage(expressionTypes, WEBSOCKET_PACKAGE_NAME)) {
+            addWebSocketAuthDesugarFunctionInvocation(functionNode, env);
+        }
+    }
+
+    boolean isValidPackage(List<BType> expressionTypes, String packageName) {
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java:106]: I this this should be something like `isDefinedInStdLibPackage`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java [hunk 2]
```diff
@@ -151,25 +162,90 @@
...
+        List<BLangStatement> statements = getFunctionBodyStatementList(functionNode);
+        statements.add(0, result);
+
+        BVarSymbol resultSymbol = new BVarSymbol(0, names.fromIdNode(result.var.name), env.enclPkg.packageID,
+                                                 result.var.type, resourceNode.symbol, pos, VIRTUAL);
+        resourceNode.symbol.scope.define(resultSymbol.name, resultSymbol);
+        result.var.symbol = resultSymbol;
+    }
+
+    void addWebSocketAuthDesugarFunctionInvocation(BLangFunction functionNode, SymbolEnv env) {
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java:179]: It looks like these 2 methods do the kind of same things, shall we try to extract common code into methods and re-use them in this 2 methods.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java [hunk 3]
```diff
@@ -151,25 +162,90 @@
...
+    private BPackageSymbol getPackageSymbol(SymbolEnv env, String packageName) {
+        // This resolves the package symbol when the code have an import relevant to the particular service
         for (BLangImportPackage pkg : env.enclPkg.imports) {
-            if (pkg.symbol.pkgID.orgName.value.equals(ORG_NAME) && pkg.symbol.pkgID.name.value.equals(PACKAGE_NAME)) {
+            if (pkg.symbol.pkgID.orgName.value.equals(ORG_NAME) && pkg.symbol.pkgID.name.value.equals(packageName)) {
                 return pkg.symbol;
             }
         }
+        // This resolves the package symbol when the code is at a submodule of the module which have the service
+        // definition. In that case there is no any import relevant to the particular service.
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java:228]: This resolves the package symbol when the code is at a submodule of the module which have the listener definition.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java (hunk 1 above)
Comment (ldclakmal) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: Yes. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31045/commits/f29c2b1d5fdd13cce00ba38fb2be681a07a5ea65

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java (hunk 2 above)
Comment (ldclakmal) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: Agree. Updated into a single method and there will be few additional steps if the package is HTTP. Else, everything will be executed.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java (hunk 3 above)
Comment (ldclakmal) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: Updated with https://github.com/ballerina-platform/ballerina-lang/pull/31045/commits/f29c2b1d5fdd13cce00ba38fb2be681a07a5ea65

**Review by rdhananjaya — APPROVED** [review]
Changes look good to me

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**